lib/plugins/helper/toc.ts

```diff
@@ -102,4 +104,27 @@ function tocHelper(str: string, options: Options = {}) {
   return result;
 }
 
+function getAndTruncateTocObj(str: string, options: {min_depth: number, max_depth: number}, max_items: number) {
+  let data = tocObj(str, { min_depth: options.min_depth, max_depth: options.max_depth });
+
+  if (data.length === 0) {
+    return data;
+  }
+  if (max_items < 1 || max_items === Infinity) {
+    return data;
+  }
+
+  const levels = data.map(item => item.level);
+  const min = Math.min(...levels);
+  const max = Math.max(...levels);
+
+  for (let currentLevel = max; data.length > max_items && currentLevel > min; currentLevel--) {
+    data = data.filter(item => item.level < currentLevel);
+  }
+
+  data = data.slice(0, max_items);
```

Comment:
This is not necessary.

Reply:
It seems necessary because if we delete this code, the tests for [`max_items - result of h1 was truncated`](https://github.com/hexojs/hexo/pull/5487/files#diff-2625d80ed02e147c8693d6cb4c52d0477bd1e299518df8097b574ef14e6987f9R666) will fail.